Userland/Applications/PixelPaint/Tools/BrushTool.cpp

```diff
@@ -199,16 +220,32 @@ NonnullRefPtr<Gfx::Bitmap> BrushTool::build_cursor()
     painter.draw_line({ scaled_size, scaled_size - 5 }, { scaled_size, scaled_size + 5 }, Color::LightGray, 3);
     painter.draw_line({ scaled_size - 5, scaled_size }, { scaled_size + 5, scaled_size }, Color::MidGray, 1);
     painter.draw_line({ scaled_size, scaled_size - 5 }, { scaled_size, scaled_size + 5 }, Color::MidGray, 1);
-    aa_painter.draw_ellipse(Gfx::IntRect(0, 0, containing_box_size, containing_box_size), Color::LightGray, 1);
+
+    // If no ImageEditor is present, we cannot bind the ellipse within the editor. Then we will just draw the ellipse.
+    if (!m_editor) {
+        aa_painter.draw_ellipse(Gfx::IntRect(0, 0, containing_box_size, containing_box_size), Color::LightGray, 1);
+    } else if (m_current_tool_position.x() + scaled_size <= m_editor->width() && m_current_tool_position.y() + scaled_size <= m_editor->height() && m_current_tool_position.x() - scaled_size > 0 && m_current_tool_position.y() - scaled_size > 0) {
+        Color color = m_editor->color_for(GUI::MouseButton::Primary);
```

Comment:
I'm not a fan of using just the primary color here and even do a fill of the ellipse. If the user presses the Secondary butten than a different color is used to draw onto the image. And if the cursor gets filled with the possible wrong color you cant even see what you are overdrawing. Assuming a alpha of 100 might also not be the correct assumtion.
I would definatly prefere not to use a fill mode and assume a color that might be uses. Having the LightGray unfilled ellipse dosn't hide anything and gives a good impression what are will be painted - especially when you draw masks. But thats just my personal preference.